.github/workflows/pre-commit.yaml

```diff
@@ -10,29 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Setup python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.x
-
-      - name: Install pre-commit
-        run: python -m pip install pre-commit
-        shell: bash
-
-      - name: Freeze dependencies
-        run: python -m pip freeze --local
-        shell: bash
-
-      - name: Cache pre-commit environment
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.cache/pre-commit
-            ~/.cache/R
-          key: pre-commit-3-${{ env.pythonLocation }}-${{ hashFiles('.pre-commit-config.yaml') }}
-
-      - name: Run pre-commit
-        run: pre-commit run --show-diff-on-failure --color=always --all-files
-        shell: bash
+      - name: Run pre-commit checks
+        uses: ccao-data/actions/pre-commit@troubleshoot-model-res-avm-hooks
```

Comment:
**[Suggestion, non-blocking]** If you merge https://github.com/ccao-data/actions/pull/27 before merging this PR, you should feel free to adjust this line back to `main` before merging.